Model/Serverside/Data/Customer.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright © PureClarity. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+namespace Pureclarity\Core\Model\Serverside\Data;
+
+use Magento\Customer\Model\Session as CustomerSession;
+
+/**
+ * Serverside Customer handler, gets customer info from session and determines if an event needs to be fired
+ */
+class Customer
+{
+    /** @var CustomerSession */
+    private $customerSession;
+
+    /**
+     * @param CustomerSession $customerSession
+     */
+    public function __construct(
+        CustomerSession $customerSession
+    ) {
+        $this->customerSession = $customerSession;
+    }
+
+    /**
+     * Checks to see if the customer has changed, to determine if an event gets fired
+     *
+     * @return array
+     */
+    public function checkCustomer()
+    {
+        $lastDetails = $this->customerSession->getPureclarityLastCustomerDetailsHash();
+        $details = $this->getCustomerDetails();
+        $send = false;
+
+        if (empty($details) || $details['hash'] !== $lastDetails) {
+            $this->customerSession->setPureclarityLastCustomerDetailsHash($details['hash']);
```

Comment:
if there are no details - I don't think we want to send?
Might want to check if we did before, and therefore we need to do a logout?

Reply:
as with the other one, it wont ever trigger empty($details) as it returns a fixed array, will remove